wgsl/index.bs

```diff
@@ -14844,30 +14844,104 @@ Note: For unpacking snorm values, the normalized floating point result is in the
-synchronization function.
+That is, all synchronization functions, and affect memory and atomic operations
+are ordered [[#program-order|program order]] relative to the synchronization
+function.
 Additionally, the affected memory and atomic operations program-ordered before
 the synchronization function must be visible to all other threads in the
 workgroup before any affected memory or atomic operation program-ordered after
 the synchronization function is executed by a member of the workgroup.
-All synchronization functions use the `Workgroup` [=memory scope=].
-All synchronization functions have a `Workgroup` [=execution scope=].
-All synchronization functions [=shader-creation error|must=] only be used in the [=compute=] shader
-stage.
 
-`storageBarrier` affects memory and atomic operations in the [=address spaces/storage=] address space.
+All synchronization functions use the `Workgroup` [=memory scope=].<br>
+All synchronization functions have a `Workgroup` [=execution scope=].<br>
+All synchronization functions [=shader-creation error|must=] only be used in
+the [=compute=] shader stage.
+
+### `storageBarrier` ### {#storageBarrier-builtin}
+
+<table class='data builtin'>
+  <tr algorithm="storageBarrier">
+    <td style="width:10%" style="width:10%">Overload
+    <td class="nowrap"><xmp highlight=rust>
+fn storageBarrier()
+</xmp>
+  <tr>
+    <td>Description
+    <td>Executes a [=control barrier=] synchronization function that affects
+    memory and atomic operations in the [=address spaces/storage=] address
+    space.
+</table>
+
+### `workgroupBarrier` ### {#workgroupBarrier-builtin}
+
+<table class='data builtin'>
+  <tr algorithm="workgroupBarrier">
+    <td style="width:10%" style="width:10%">Overload
+    <td class="nowrap"><xmp highlight=rust>
+fn workgroupBarrier()
+</xmp>
+  <tr>
+    <td>Description
+    <td>Executes a [=control barrier=] synchronization function that affects
+    memory and atomic operations in the [=address spaces/workgroup=] address
+    space.
+</table>
+
+### `workgroupBroadcast` ### {#workgroupBroadcast-builtin}
+
+<table class='data builtin'>
+  <tr algorithm="workgroupBroadcast-id">
+    <td style="width:10%" style="width:10%">Overload
+    <td class="nowrap"><xmp highlight=rust>
+fn workgroupBroadcast(value : T, id : vec3<u32>) -> T
+</xmp>
+  <tr>
+    <td>Parameterization
+    <td>`T` is a [=type/concrete=] [=plain type=] with a [=fixed footprint=]
+    that does not contain any [=atomic types=]
+  <tr>
+    <td>Description
+    <td>Returns the value of `value` from the invocation `id` to all
+    invocations in the workgroup.
+    The return value is [=uniform value|uniform=].
+
+    Executes a [=control barrier=] synchronization function that affects
+    memory and atomic operations in the [=address spaces/workgroup=] address
+    space.
+
+    Each static instance of this built-in allocates
+    [=roundUp=]`(`[=sizeof=]`(T), 16)` bytes of workgroup memory.
```

Comment:
I think the parameter order for our definition of `roundUp` is the other way around.